nextcord/embeds.py

```diff
@@ -360,7 +368,11 @@ def footer(self) -> _EmbedFooterProxy:
         return EmbedProxy(getattr(self, "_footer", {}))  # type: ignore
 
     def set_footer(
-        self: E, *, text: MaybeEmpty[Any] = EmptyEmbed, icon_url: MaybeEmpty[Any] = EmptyEmbed
+        self: E,
+        *,
+        file: Optional[File] = None,
+        text: MaybeEmpty[Any] = EmptyEmbed,
+        icon_url: MaybeEmpty[Any] = EmptyEmbed,
```

Comment:
Perhaps EmptyEmbed should be used for consistency, and maybe name it icon_file to match with icon_url?

```suggestion
        text: MaybeEmpty[Any] = EmptyEmbed,
        icon_url: MaybeEmpty[Any] = EmptyEmbed,
        icon_file: MaybeEmpty[File] = EmptyEmbed,
```

Reply:
I was thinking `file` for consistency with the other 3 messages

Reply:
In the other places there is `url` and `file`, since this has `icon_url` instead, then `icon_file` would match. Naming it wirh icon makes it more clear what the "footer file" / "author file" is for.